sdks/python/apache_beam/typehints/sharded_key_type.py

```diff
@@ -25,8 +25,12 @@
 from apache_beam.typehints.typehints import match_type_variables
 from apache_beam.utils.sharded_key import ShardedKey
 
+from future.utils import with_metaclass
 
-class ShardedKeyTypeConstraint(typehints.TypeConstraint):
+
+class ShardedKeyTypeConstraint(with_metaclass(typehints.GetitemConstructor,
```

Comment:
This seems to be casuing lint failure. I was following the `WindowedValue` example:

https://github.com/apache/beam/blob/30f9a607509940f78459e4fba847617399780246/sdks/python/apache_beam/typehints/typehints.py#L1050

Reply:
What kind of lint errors are you getting?

Reply:
I don't know what `with_metaclass` does. You could add the `__getitem__` implementation directly if it's causing issues.

Reply:
`GetItemConstructor` basically defines `__getitem__`. I tried getting rid of it and defining `__getitem__` directly but I got the error
```
TypeError: 'type' object is not subscriptable
```

Reply:
This works:
```py
  def __class_getitem__(cls, py_type):
    return cls(py_type)
```

The alternative is to use a wrapper object instance such as `IterableHint` that has `__getitem__`, and then define `Iterable = IterableHint()`.

Reply:
Thanks for the tip!!! Updated the code to define `__class_getitem__` directly in `ShardedKeyTypeConstraint`.